packages/components/src/components/table/Table.tsx

```diff
@@ -0,0 +1,191 @@
+  };
+
+  const getRowsPerPages = (select) => {
+    if (select.value) {
+      return select.value;
+    } else {
+      return rowsPerPage;
+    }
+  };
+
+  const getPage = (currentIndex, pageSize, array) => {
+    const startIndex = (currentIndex - 1) * pageSize;
+    const endIndex = startIndex + pageSize;
+    return array.slice(startIndex, endIndex);
+  };
+
+  const getTotalPages = (pageSize, array) => {
+    const result = array.length / pageSize;
+    return Math.ceil(result);
+  };
+
+  const hasNextPage = (currentPage, pageSize, array) => {
+    const nextPageIndex = currentPage * pageSize;
+    return nextPageIndex < array.length;
+  };
+
+  const renderPagination = () => {
+    return (
+      <div className="tk-table-pagination">
+        <Icon
+          className={
+            currentPage === 1 ? 'tk-table-button-disabled' : 'tk-table-button'
+          }
+          iconName="chevron-left"
+          disabled={currentPage === 1}
+          onClick={() => setCurrentPage(1)}
+        />
+        <Icon
+          className={
+            currentPage <= 1 ? 'tk-table-button-disabled' : 'tk-table-button'
+          }
+          iconName="left"
+          disabled={currentPage <= 1}
+          onClick={() => setCurrentPage(currentPage - 1)}
+        />
+        <p>Rows per page: </p>
+        <Dropdown
+          className="tk-table-dropdown"
+          size="small"
+          onChange={handleSelectChange}
+          placeHolder={rowsPerPage.toString()}
+          options={[
+            {
+              label: '5',
+              value: 5,
+            },
+            {
+              label: '10',
+              value: 10,
+            },
+            {
+              label: '25',
+              value: 25,
+            },
+            {
+              label: '50',
+              value: 50,
+            },
+            {
+              label: '100',
+              value: 100,
+            },
+          ]}
+        />
+        <p>
+          Page {currentPage} of {totalPages}
+        </p>
+        <Icon
+          className={nextPage ? 'tk-table-button' : 'tk-table-button-disabled'}
+          iconName="right"
+          disabled={!nextPage}
+          onClick={() => setCurrentPage(currentPage + 1)}
+        />
+        <Icon
+          className={nextPage ? 'tk-table-button' : 'tk-table-button-disabled'}
+          iconName="chevron-right"
+          disabled={!nextPage}
+          onClick={() => setCurrentPage(totalPages)}
+        />
+      </div>
+    );
+  };
+
+  const totalEntries = data.length;
+  const page = getPage(currentPage, getRowsPerPages(select), data);
+  const totalPages = getTotalPages(getRowsPerPages(select), data);
+  const nextPage = hasNextPage(currentPage, getRowsPerPages(select), data);
+
+  return (
+    <>
+      <table className="tk-table">
+        <thead>
+          <tr>
+            <th className="tk-table-checkbox">
+              {showCheckbox && <Checkbox label="" name="simple-checkbox" />}
```

Comment:
isn't it better if we don't generate the th element if not needed? 
the condition of showCheckbox should maybe be in line 159